src/multi.rs

```diff
@@ -340,7 +371,23 @@ impl MultiState {
     }
 
     fn len(&self) -> usize {
-        self.draw_states.len() - self.free_set.len()
+        self.elements.len() - self.free_set.len()
+    }
+}
+
+#[derive(Default)]
+struct MultiStateElement {
```

Comment:
So with `draw_state: Option<_>` and `is_zombie: bool` we have 4 different states. Do we need all 4? Maybe that would be a useful thing to explain in a comment?

I was surprised `Weak` implements `Default` -- it clearly works but it feels a little ugly to me. If we can collapse some of the states maybe there's something better we can do?

Reply:
Well, only three states are necessary:

1. On `MultiProgress::internalize` (and before draw) `draw_state` will be `None` and `is_zombie` will be `false`.
2. On draw, `draw_state` will be `Some(..)` and `is_zombie` will be `false`.
3. When a zombie is detected, `is_zombie` will become `true`. Normally `draw_state` will be `Some(..)` but I suppose it's possible for a pb to be briefly in zombie state if it was never drawn (i.e. you immediately drop it after adding to the `MultiProgress`). 

I'll add a comment to this effect.

Reply:
One option would be to have an enum, but I think that would make the code even uglier.

Reply:
By "suppose it's possible", I think you're suggesting that we actually do have 4 states?

Reply:
Yes, there are technically four states. But state 3 and 4 (with 4 being "is_zombie == true" and "draw_state is None") are handled the same.